dae/dae/pheno/pheno_import.py

```diff
@@ -0,0 +1,824 @@
+) -> pd.DataFrame:
+    """
+    Read a pedigree file into a pandas DataFrame
+
+    Also imports the pedigree data into the database.
+    """
+    ped_df = FamiliesLoader.flexible_pedigree_read(
+        pedigree_filepath, enums_as_values=True,
+    )
+
+    with connection.cursor() as cursor:
+        cursor.execute(
+            "INSERT INTO family "
+            "SELECT DISTINCT family_id FROM ped_df",
+        )
+        cursor.execute(
+            "INSERT INTO person "
+            "SELECT family_id, person_id, "
+            "role, status, sex, sample_id FROM ped_df ",
+        )
+    return ped_df
+
+
+def write_results(
+    connection: duckdb.DuckDBPyConnection,
+    instrument_tables: dict[str, Any],
+    ped_df: pd.DataFrame,
+) -> None:
+    """Write imported data into duckdb as measure value tables."""
+    with connection.cursor() as cursor:
+        for instrument_name, table in instrument_tables.items():
+            output_table_df = pd.DataFrame(
+                table).transpose().rename_axis("person_id").reset_index()
+
+            output_table_df = output_table_df.merge(
+                ped_df[["person_id", "family_id", "role", "status", "sex"]],
+                on="person_id", how="left",
+            )
+
+            cols = output_table_df.columns.tolist()
+
+            # Reorder columns so that pedigree columns are first
+            output_table_df = \
+                output_table_df[cols[0:1] + cols[-4:] + cols[1:-4]]
+
+            output_table_df.drop(
+                output_table_df[
+                    ~output_table_df["person_id"].isin(ped_df["person_id"])
+                ].index,
+            )
+
+            table_name = safe_db_name(f"{instrument_name}_measure_values")
+            cursor.execute(
+                textwrap.dedent(f"""
+                    INSERT INTO instrument VALUES
+                    ('{instrument_name}', '{table_name}')
+                """),
+            )
+            cursor.execute(
+                textwrap.dedent(f"""
+                    CREATE TABLE {table_name} AS
+                    SELECT * FROM output_table_df
+                """),
+            )
+
+
+def load_descriptions(
+    description_path: str | None,
+) -> Callable | None:
+    """Load measure descriptions."""
+    if not description_path:
+        return None
+    assert os.path.exists(
+        os.path.abspath(description_path),
+    ), description_path
+
+    data = pd.read_csv(description_path, sep="\t")
+
+    class DescriptionDf:
+        """Phenotype database support for measure descriptions."""
+
+        def __init__(self, desc_df: pd.DataFrame):
+            self.desc_df = desc_df
+            assert all(
+                col in list(desc_df)
+                for col in [
+                    "instrumentName",
+                    "measureName",
+                    "measureId",
+                    "description",
+                ]
+            ), list(desc_df)
```

Comment:
Taking out ``list(desc_df)`` into its own variable might be a (tiny) optimization here